src/core/cgroup.c

```diff
@@ -4653,7 +4652,7 @@ int unit_get_cpu_usage(Unit *u, nsec_t *ret) {
         if (!UNIT_CGROUP_BOOL(u, cpu_accounting))
                 return -ENODATA;
 
-        r = unit_get_cpu_usage_raw(u, &ns);
+        r = unit_get_cpu_usage_raw(u, crt, &ns);
         if (r == -ENODATA && crt->cpu_usage_last != NSEC_INFINITY) {
```

Comment:
This code is not reached when the unit shuts down, because `crt->cgroup_path` is `NULL`.
That check needs to be removed above.  That is save because `unit_get_cpu_usage_raw()` will just return `-ENODATA` in that case and the fallback here will return the cached value as expected.